app/bundles/QueueBundle/MauticQueueBundle.php

```diff
@@ -11,11 +11,82 @@
 
 namespace Mautic\QueueBundle;
 
+use Leezy\PheanstalkBundle\DependencyInjection\LeezyPheanstalkExtension;
+use Mautic\QueueBundle\Queue\QueueProtocol;
+use OldSound\RabbitMqBundle\DependencyInjection\Compiler\RegisterPartsPass;
+use OldSound\RabbitMqBundle\DependencyInjection\OldSoundRabbitMqExtension;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
-/**
- * Class MauticQueueBundle.
- */
 class MauticQueueBundle extends Bundle
 {
+    /**
+     * @var array
+     */
+    private $localParams;
+
+    public function __construct(array $localParams)
+    {
+        $this->localParams = $localParams;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function build(ContainerBuilder $container)
+    {
+        parent::build($container);
+
+        $queueProtocol = $this->getQueueProtocol();
+
+        if (QueueProtocol::RABBITMQ === $this->getQueueProtocol()) {
+            $container->addCompilerPass(new RegisterPartsPass());
+        }
+
+        if ($queueProtocol && file_exists(__DIR__.'/Config/'.$queueProtocol.'.php')) {
+            include __DIR__.'/Config/'.$queueProtocol.'.php';
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContainerExtension()
+    {
+        if (null === $this->extension) {
+            $extension       = $this->createContainerExtension();
+            $this->extension = $extension;
+        }
+
+        return $this->extension ?: null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createContainerExtension()
+    {
+        $queueProtocol = $this->getQueueProtocol();
+
+        if (!$queueProtocol) {
+            return null;
+        }
+
+        if (QueueProtocol::RABBITMQ === $queueProtocol) {
+            return new OldSoundRabbitMqExtension();
+        }
+
+        if (QueueProtocol::BEANSTALKD === $queueProtocol) {
+            return new LeezyPheanstalkExtension();
+        }
```

Comment:
Maybe ditch lines 71-73 and add a `return null;` here at the end.

Reply:
I prefer returning strictly same type.

Reply:
If the queue protocol is set to something else besides beanstalk or rabbitmq then it won't return anything at all